classes/Context.php

```diff
@@ -423,30 +422,8 @@ public function getTranslatorFromLocale($locale)
         }
 
         $adminContext = defined('_PS_ADMIN_DIR_');
-        $translator->addLoader('xlf', new XliffFileLoader());
-
-        $sqlTranslationLoader = new SqlTranslationLoader();
-        if (null !== $this->shop) {
-            $sqlTranslationLoader->setTheme($this->shop->theme);
-        }
-
-        $translator->addLoader('db', $sqlTranslationLoader);
-        $notName = $adminContext ? '^Shop*' : '^Admin*';
-
-        $finder = Finder::create()
-            ->files()
-            ->name('*.' . $locale . '.xlf')
-            ->notName($notName)
-            ->in($this->getTranslationResourcesDirectories());
-
-        foreach ($finder as $file) {
-            list($domain, $locale, $format) = explode('.', $file->getBasename(), 3);
-
-            $translator->addResource($format, $file, $locale, $domain);
-            if (!$this->language instanceof PrestashopBundle\Install\Language) {
-                $translator->addResource('db', $domain . '.' . $locale . '.db', $locale, $domain);
-            }
-        }
+        $theme = $this->shop !== null ? $this->shop->theme : null;
+        (new TranslatorLanguageLoader($adminContext))->loadLanguage($translator, $locale, $theme);
```

Comment:
Following my comment on DataLang, you may extract further logic from this method and create a static factory which would replace `TranslatorLanguageLoader` which would no longer be a necessary as a separate feature:

```php
$options = (new TranslationFactoryOptions())
  ->setIsBackoffice(true)
  ->setThemeName(null) // not needed when setIsBackoffice is true
  ->setCleanCache(true);
$newTranslator = TranslationFactory::buildTranslator($locale, $options);
```

Reply:
Nevermind, not necessary